src/robot_launch_files/odometer.py

```diff
@@ -0,0 +1,194 @@
+#! /usr/bin/python
+
+from __future__ import print_function
+
+import os
+import socket
+import time
+import csv
+from fnmatch import fnmatch
+import rospy
+from math import sqrt, pow, pi
+from tf.transformations import euler_from_quaternion
+
+from nav_msgs.msg import Odometry
+
+DEFAULT_PATH = "~/MEGA/odometer"
+DEFAULT_FILENAME = 'odometer'
+EXT = '.csv'
+ROUND_LEVEL = 5
+
+
+class Odometer:
+    def __init__(self, path=DEFAULT_PATH, filename=DEFAULT_FILENAME):
+        if fnmatch(path, "~*"):  # if path is in home folder
+            path = os.path.expanduser(path)
+        path = os.path.abspath(path)  # returns abs path, also when path is already abs.
+
+        hostname = socket.gethostname()
+        date = time.strftime("%Y_%m_%d")
+
+        hostfolderpath = os.path.join(os.path.expanduser(path), hostname.lower())
```

Comment:
This is quite sneaky. When I have parameters path and filename I expect to find path/filename on my computer, not path/hostname/filename.

Since you're specifying a parameter file for amigo and sergio anyway I would just leave the whole hostname feature out and specify it in the yaml file.